subsys/bluetooth/host/cs.c

```diff
@@ -241,6 +242,12 @@ void bt_hci_le_cs_read_remote_fae_table_complete(struct net_buf *buf)
 	bt_conn_unref(conn);
 }
 
+int bt_le_cs_test_cb_register(struct bt_le_cs_test_cb cb)
```

Comment:
what do you think of a BT_CHANNEL_SOUNDING_TEST ?

Reply:
> what do you think of a BT_CHANNEL_SOUNDING_TEST ?

Would it have `depends on BT_TESTING`? At least until now I believe we've grouped all testing related Kconfig options behind `BT_TESTING`. The bluetooth tester app (used for qualification) enables `BT_TESTING` so it shouldn't be an issue there. The Bluetooth shell also enables it for some use cases (see e.g. `tests/bluetooth/shell/audio.conf`)

Reply:
I was thinking of doing the same as with BT_ISO_TEST_PARAMS (no strong dependencies). No strong opinion however, I'm not so familiar with the use-cases for BT_TESTING

Reply:
`BT_ISO_TEST_PARAMS` doesn't seem to be guarding for any public APIs (functions), so I don't think it's quite the same. *If* we will tie these CS features to `BT_TESTING` I don't think we necessarily need a separate option, since it doesn't really provide added value. @jori-nordic @alwa-nordic any thoughts here?

Reply:
the nice part of BT_CHANNEL_SOUNDING_TEST is I think it makes it easier to add a corresponding controller kconfig, which can then have something like: `default n if !BT_CHANNEL_SOUNDING_TEST` (which can be used to remove controller code implementing these commands)

by default the test commands need to be included in the controller as they're mandatory by spec

Reply:
Sure, as long as you don't forget that the controller doesn't have to be part of the same build as the host, e.g. if there's a physical transport between them and they're on different cores.

Reply:
Yes, that's true.

Reply:
@jhedberg In any case I've pushed what I imagined this could look like, does it seem ok?

Reply:
> @jhedberg In any case I've pushed what I imagined this could look like, does it seem ok?

I was thinking that this might still `depends on BT_TESTING` as well. @alwa-nordic any thoughts on this?

Reply:
I have some confidence this does not belong in BT_TESTING. My understanding is we use BT_TESTING to do white-box testing in CI. It exposes some interfaces that allow observing or altering something about internals of the stack. These interfaces must be considered implementation-leaking and very unstable.

`config BT_TESTING` says "This option enables *custom* Bluetooth testing interface." (emphasis mine).

If this is an HCI function, then it's not custom. And if this is useful for testing interoperability and RF environment, then it may be useful for end-users. And I don't want end-users to rely on BT_TESTING.

Reply:
Said plain: I would like BT_TESTING to be owned by, and only available to, the Host (or at least only in-tree in Zephyr).